python/hard_clone_workspace.py

```diff
@@ -0,0 +1,208 @@
+            Csv(file_path=tsv, delimiter="\t").create_tsv_from_list_of_dicts(
+                header_list=headers,
+                list_of_dicts=updated_table_metadata
+            )
+            tsv_to_upload.append(tsv)
+        return tsv_to_upload
+
+
+class CopyFilesToDestWorkspace:
+    def __init__(self, src_bucket: str, dest_bucket: str, workers: int, extensions_to_ignore: list[str] = []):
+        self.src_bucket = src_bucket
+        self.dest_bucket = dest_bucket
+        self.extensions_to_ignore = extensions_to_ignore
+        self.gcp_cloud_functions = GCPCloudFunctions()
+        self.workers = workers
+
+    def run(self) -> None:
+        GCPCloudFunctions()
+        logging.info(f"Getting all files from source bucket {self.src_bucket}")
+        list_bucket_contents = self.gcp_cloud_functions.list_bucket_contents(
+            bucket_name=self.src_bucket,
+            file_extensions_to_ignore=self.extensions_to_ignore
+        )
+        files_to_copy = [
+            {
+                "source_file": file_info['path'],
+                "full_destination_path": file_info['path'].replace(self.src_bucket, self.dest_bucket)
+            }
+            for file_info in list_bucket_contents
+        ]
+        if not files_to_copy:
+            logging.info("No files to copy")
+            return
+        else:
+            logging.info(f"Copying {len(files_to_copy)} files to destination bucket {self.dest_bucket}")
+            self.gcp_cloud_functions.multithread_copy_of_files_with_validation(
+                files_to_move=files_to_copy,
+                workers=self.workers,
+                max_retries=5
+            )
+
+
+class UpdateWorkspaceAcls:
+    def __init__(self, src_workspace: TerraWorkspace, dest_workspace: TerraWorkspace):
+        self.src_workspace = src_workspace
+        self.dest_workspace = dest_workspace
+
+    def run(self) -> None:
+        # Get the source workspace ACLs and entities
+        src_workspace_acls = self.src_workspace.get_workspace_acl()
+        # Convert the source workspace ACLs to a list of dictionaries
+        src_workspace_acls_list = [
+            {
+                "email": user,
+                "accessLevel": user_acl["accessLevel"],
+                "canShare": user_acl["canShare"],
+                "canCompute": user_acl["canCompute"]
+            }
+            for user, user_acl in src_workspace_acls["acl"].items()
+        ]
+        self.dest_workspace.update_multiple_users_acl(acl_list=src_workspace_acls_list)
+
+
+if __name__ == '__main__':
+    args = get_args()
+    source_billing_project = args.source_billing_project
+    source_workspace_name = args.source_workspace_name
+    dest_billing_project = args.dest_billing_project
+    dest_workspace_name = args.dest_workspace_name
+    allow_already_created = args.allow_already_created
+    workers = args.workers
+    extensions_to_ignore = args.extensions_to_ignore.split(",") if args.extensions_to_ignore else None
```

Comment:
I just added a custom method/type [here](https://github.com/broadinstitute/ops-terra-utils/blob/9b4154338cc060d6a0286318950510aa80b296a4/python/gcp_workspace_table_to_dataset_ingest.py#L33-L34) and used it [here](https://github.com/broadinstitute/ops-terra-utils/blob/9b4154338cc060d6a0286318950510aa80b296a4/python/gcp_workspace_table_to_dataset_ingest.py#L56-L61), maybe we should move that to an `__init__` file and re-use it for situations like this.